File: Programowanie Obiektowe Projekt Nr 3 Python/Wolf.py
#-------------------------------------------------------------------------------
# Name:        module1
# Purpose:
#
# Author:      Kontekstowy
#
# Created:     27-05-2016
# Copyright:   (c) Kontekstowy 2016
# Licence:     <your licence>
#-------------------------------------------------------------------------------
from Animal import Animal
class Wolf(Animal):
    def __init__(self,xx,yy,world,numberx,tempSign):
        Animal.__init__(self,xx,yy,world,numberx,tempSign)
        self._initiative=5
        self._strength=9
        self._sign='W'
    def action(self,animalBoard,nx,ny,animalContainer,board):
        if animalBoard[self._x  + nx][self._y +ny] == -1:
            return 1
        if animalBoard[self._x + nx][self._y + ny] != -1:
            if animalBoard[self._x + nx][self._y + ny] == 'F':
                print("Wilk zabija lisa")
                return 2
            if animalBoard[self._x + nx][self._y + ny] == 'T':
                print("Wilka zabija zolwia")
                return 2
            if animalBoard[self._x + nx][self._y + ny] == 'W':
                if nx !=0 and ny !=0:
                    return 4
                else:
                    return 0
            if animalBoard[self._x + nx][self._y + ny] == 'A':
                print("Wilk zabija antylope")
                return 2
            if animalBoard[self._x + nx][self._y + ny] == 'H':
                print("Wilk zabija czlowieka")
                return 2
            if animalBoard[self._x + nx][self._y + ny] == '#':
                print("Wilk zjada trawe")
                return 2
            if animalBoard[self._x + nx][self._y + ny] == 'G':
                print("Wilk zjada Guarane")
                return 2
            if animalBoard[self._x + nx][self._y + ny] == 'N':
                print("Wilk zjada trujace wilcze jagody i ginie")
                return 5
            if animalBoard[self._x + nx][self._y + ny] == 'L':
                print("Wilk zjada mleczyk")
                return 2
            if animalBoard[self._x + nx][self._y + ny] == 'S':
                print("Wilk zjada owce")
                return 2

File: Programowanie Obiektowe Projekt Nr 2 Java/src/objectiveworld160553/Wolf.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package objectiveworld160553;

/**
 *
 * @author Kontekstowy
 */
public class Wolf extends Animal {

    public Wolf(int x, int y, World world) {
        super(x, y, world);
        this.name = "W";
        this.initiative = 5;
        this.strength = 9;
    }

    @Override
    public int action(char animalBoard[][], int nx, int ny, Organism me) {
        switch (animalBoard[x + nx][y + ny]) {
            case '0':
                return 1;
            case 'W':
          //      System.out.println("Spotkaly sie 2 wilki");
                return 4;
            case 'S':
                System.out.println("Wilk zabija owce");
                return 2;
            case 'N':
                System.out.println("Wilk zjada wilcze jagody i ginie");
                return 5;
            case 'T':
                System.out.println("Wilk zabija zolwia");
                return 2;
            case 'F':
                System.out.println("Wilk zabija lisa");
                return 2;
            case 'L':
                System.out.println("Wilk zjada mlecz");
                return 2;
            case '#':
                System.out.println("Wilk zjada trawe");
                return 2;
            case 'G':
                System.out.println("Wilk zjadla guaraney");
                return 2;
            case 'H':
                System.out.println("Wilk zabija czlowieka");
                return 2;
        }

        return 0;
    }
}
